cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinMapper.java

```diff
@@ -219,7 +220,11 @@ private CristinBookReport extractCristinBookReport() {
     }
 
     private String extractNumberOfPages() {
-        return extractCristinBookReport().getNumberOfPages();
+        String numberOfPages = extractCristinBookReport().getNumberOfPages();
+        if (StringUtils.isBlank(numberOfPages)) {
+            throw new InvalidCristinBookReportEntryException("numberOfPages", numberOfPages);
+        }
+        return numberOfPages;
```

Comment:
I think there is a problem here that when this error will occur the JSON fields are not actually  called BookReport and numberOfPages. I would suggest moving this check in the getter of the object. 
i.e.,
```
   public String getNumberOfPages() {
        if (Objects.nonNull(numberOfPages)) {
            return numberOfPages;
        }

        throw new InvalidCristinBookReportEntryException(NUMBER_OF_PAGES, numberOfPages);
    }
```

In this way return at least the correct name of the field and if the name of the field changes then the error message will change automatically

Reply:
In general if you have something like this in the code:
```
foo(){
var x= this.object.getX();
var y = this.object.getY();
var z = this.object.processXandY(x,y);
this.object.doodle(x)
this.object.booble(y)
```
This is an indication that the whole method belongs to the class of the obect